sparta/scripts/GenRegisterJSON.py

```diff
@@ -0,0 +1,463 @@
+            name = self.get_int_reg_name(num)
+        if group_num is RegisterGroup.INT:
+            self.gen_int_reg_defs()
+        elif group_num is RegisterGroup.FP:
+            self.gen_fp_reg_defs()
+        elif group_num is RegisterGroup.CSR:
+            self.gen_csr_reg_defs()
+            self.gen_csr_header_file()
+            self.gen_csr_bit_fields()
+        elif group_num is RegisterGroup.VEC:
+            self.gen_vec_reg_defs()
+
+    def get_int_reg_name(self, num):
+        """Returns int register name from register number"""
+        return "x"+str(num)
+
+    def get_int_reg_desc(self, num):
+        """Returns int register description from register number"""
+        return "int register "+str(num)
+
+    def gen_int_reg_defs(self):
+        """Generates integer register definitions
+        """
+        for num in range(0, self.num_regs):
+            name = self.get_int_reg_name(num)
+            alias = INT_ALIASES[name] if (name in INT_ALIASES.keys()) else []
+            fields = {}
+            if (num == 0):
+                if (self.reg_size == 4):
+                    fields["b31_00"] = self.readonly_fields["b31_00"]
+                else:
+                    fields["b63_00"] = self.readonly_fields["b63_00"]
+
+            self.reg_defs.append(self.__CreateRegDict({
+                "name":          name,
+                "num":           num,
+                "desc":          self.get_int_reg_desc(num),
+                "size":          self.reg_size,
+                "aliases":       alias,
+                "fields":        fields,
+                "initial_value": 0,
+                "enabled":       True}))
+
+    def get_fp_reg_name(self, num):
+        """Returns fp register name from register number"""
+        return "f"+str(num)
+
+    def get_fp_reg_desc(self, num):
+        """Returns fp register description from register number"""
+        return "floating point register "+str(num)
+
+    def gen_fp_reg_defs(self):
+        """Generates floating point register definitions
+        """
+
+        fields = {}
+        fields["sp"] = {"desc":     "single precision",
+                        "low_bit":  0,
+                        "high_bit": 31,
+                        "readonly": False}
+        fields["dp"] = {"desc":     "double precision",
+                        "low_bit":  0,
+                        "high_bit": 63,
+                        "readonly": False}
+
+        for num in range(0, self.num_regs):
+            name = self.get_fp_reg_name(num)
+            alias = FP_ALIASES[name] if (name in FP_ALIASES.keys()) else []
+            self.reg_defs.append(self.__CreateRegDict({
+                "name":          name,
+                "num":           num,
+                "desc":          self.get_fp_reg_desc(num),
+                "size":          8,
+                "aliases":       alias,
+                "fields":        fields,
+                "initial_value": 0,
```

Comment:
Does this need to be changed to `"x0"`?

Reply:
In the __CreateRegDict method, I remove the "initial_value" key from the dict if it is zero, else convert to hex.